sans/reduction.ipynb

```diff
@@ -210,6 +210,55 @@
     "    return (pixel_size * pixel_length) / (L2 * L2)"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def q_resolution(lam_edges, moderator, d):\n",
+    "    moderator = sc.rebin(moderator, 'wavelength', lam_edges)\n",
+    "    \n",
+    "    d_lam = lam_edges['wavelength', 1:] - lam_edges['wavelength', :-1] # bin widths\n",
+    "    lam = 0.5 * (lam_edges['wavelength', 1:] + lam_edges['wavelength', :-1]) # bin centres\n",
+    "    \n",
+    "    r2 = sc.Variable(value=4.0824829046386295/1000, unit=sc.units.m) # sample aperture radius\n",
+    "    r1 = sc.Variable(value=14.433756729740645/1000, unit=sc.units.m) # source aperture radius\n",
+    "    dr = sc.Variable(value=8.0/1000, unit=sc.units.m) # virtual ring width on detector\n",
+    "    \n",
+    "    l2 = sc.neutron.l2(d)\n",
+    "    l1 = sc.Variable(value=5.0, unit=sc.units.m) # should be sc.neutron.l1(d) = 25.6m ?\n",
+    "    theta = sc.neutron.scattering_angle(d)  \n",
+    "\n",
+    "    inv_l3 = (l1 + l2) / (l1*l2)\n",
+    "    \n",
+    "    # Terms in Mildner and Carpenter equation. See https://docs.mantidproject.org/nightly/algorithms/TOFSANSResolutionByPixel-v1.html\n",
+    "    a1 = (r1/l1)*(r1/l1) * 3.0\n",
+    "    a2 = (r2*inv_l3)*(r2*inv_l3) * 3.0\n",
+    "    a3 = (dr/l2) * (dr/l2) \n",
+    "    q = 4.0 * np.pi * sc.sin(theta) * sc.reciprocal(lam) # keep in wav dim to prevent broadcast\n",
+    "    \n",
+    "    h = 6.62607004E-34\n",
+    "    m = 1.6749E-27\n",
+    "    k = sc.Variable(value=h/m, unit=(sc.units.m * sc.units.m)/sc.units.s) \n",
+    "    \n",
+    "    moderator.data.values *= 1e-6 # us to seconds\n",
+    "    moderator.data.unit = sc.units.s # fixup moderator units to microseconds. Previously dimensionless\n",
+    "    \n",
+    "    # Calcualte moderator spread\n",
+    "    s = moderator.data * sc.reciprocal(l1 + l2) # times to inv velocity\n",
+    "    s *= k # inv velocity to wavelength\n",
+    "    s.unit = sc.units.angstrom\n",
+    "    s *= 1e10\n",
+    "    \n",
+    "    std_dev_lam = (d_lam)/12 + s\n",
+    "    std_dev_lam *= sc.reciprocal(lam)\n",
+    "    f = (4 * np.pi * np.pi) * sc.reciprocal(12 * lam * lam)\n",
+    "   \n",
+    "    return sc.DataArray(sc.sqrt(f * (a1 + a2 + a3)) + (q * std_dev_lam), coords={'wavelength':lam, 'spectrum':d.coords['spectrum'].copy()})\n",
```

Comment:
- Missing square on `q`.
- Parenthesis of `sqrt` end too early. But probably we should just return `sigma_Q^2` anyway, so remove the sqrt?

Reply:
Yes, removing sq then sqrt was an optimisation (same result), but probably didn't improve performance much. Returning sq would also be OK. Also given the discussion about how to apply to q1d, I might need to think about the boundaries of this function - as in is returning a mantid like result that might not be necessary.

Reply:
It cannot be the same result, aren't the parenthesis in the wrong place?